conftest.py

```diff
@@ -71,10 +102,137 @@ def get_software():
 dask.config.set(
     {
         "coiled.account": "dask-engineering",
-        "coiled.software": get_software(),
+        "coiled.software": get_coiled_software_name(),
     }
 )
 
+COILED_RUNTIME_VERSION = get_coiled_runtime_version()
+
+
+# ############################################### #
+#            BENCHMARKING RELATED                 #
+# ############################################### #
+
+DB_NAME = os.environ.get("DB_NAME", "benchmark.db")
+
+if os.environ.get("GITHUB_SERVER_URL"):
+    WORKFLOW_URL = "/".join(
+        [
+            os.environ.get("GITHUB_SERVER_URL", ""),
+            os.environ.get("GITHUB_REPOSITORY", ""),
+            "actions",
+            "runs",
+            os.environ.get("GITHUB_RUN_ID", ""),
+        ]
+    )
+else:
+    WORKFLOW_URL = None
+
+
+@pytest.fixture(scope="session")
+def benchmark_db_engine(pytestconfig, tmp_path_factory):
+    if not pytestconfig.getoption("--benchmark"):
+        yield
+    else:
+        engine = sqlalchemy.create_engine(f"sqlite:///{DB_NAME}", future=True)
+
+        # get the temp directory shared by all workers
+        root_tmp_dir = tmp_path_factory.getbasetemp().parent
+        lock = root_tmp_dir / (DB_NAME + ".lock")
+
+        # Create the db if it does not exist already.
+        with filelock.FileLock(lock):
+            if not os.path.exists(DB_NAME):
+                with engine.connect() as conn:
+                    conn.execute(sqlalchemy.text("VACUUM"))
+
+        # Run migrations if we are not up-to-date.
+        with filelock.FileLock(lock):
+            current = subprocess.check_output(["alembic", "current"], text=True)
+            if "(head)" not in current:
+                p = subprocess.run(["alembic", "upgrade", "head"])
+                p.check_returncode()
+
+        yield engine
+
+
+@pytest.fixture(scope="function")
+def benchmark_db_session(benchmark_db_engine):
+    if not benchmark_db_engine:
+        yield
+    else:
+        with Session(benchmark_db_engine, future=True) as session, session.begin():
+            yield session
+
+
+@pytest.fixture(scope="function")
+def test_run_benchmark(benchmark_db_session, request, testrun_uid):
+    if not benchmark_db_session:
+        yield
+    else:
+        node = request.node
+        run = TestRun(
+            session_id=testrun_uid,
+            name=node.name,
+            originalname=node.originalname,
+            path=str(node.path.relative_to(TEST_DIR)),
+            dask_version=dask.__version__,
```

Comment:
Should we also include `distributed_version`?

Reply:
I was presuming that they would always be identical (otherwise we have bigger problems).

Do you think there is a use-case for tracking them separately? If so, now would be the time to add it.

Reply:
Ah, that's a good point. I can imagine a future where we decouple `dask` and `distributed` release cycles, though that could be far in the future. 

Alternatively, doesn't `coiled_runtime_version` capture both the `dask` and `distributed` versions?

Reply:
> Alternatively, doesn't coiled_runtime_version capture both the dask and distributed versions?

Sort of -- this is used more as a unit of categorizing different runs for comparisons. That is to say, this is either a version number, or `"latest"`, or `"upstream"`. So it's a little loose, but important for determining which timeseries a particular run goes into.

Reply:
Personally, I like capturing it even if we don't expect them to differ at all. When something is wrong, it's nice to be able to confirm that they did in fact match up.

Reply:
Okay, with n=2, I've added the distributed version